op-node/eth/account_proof.go

```diff
@@ -21,24 +27,68 @@ type AccountResult struct {
 	CodeHash    common.Hash    `json:"codeHash"`
 	Nonce       hexutil.Uint64 `json:"nonce"`
 	StorageHash common.Hash    `json:"storageHash"`
-	// storageProof field is ignored, we only need to proof the account contents,
-	// we do not access any individual storage values.
+
+	// Optional
+	StorageProof []StorageProofEntry `json:"storageProof,omitempty"`
 }
 
```

Comment:
You might be able to use a much lighter type than `memorydb` for the storage proof

```go
// Custom type to write the generated proof to
type proofList [][]byte

func (n *proofList) Put(key []byte, value []byte) error {
	*n = append(*n, value)
	return nil
}

func (n *proofList) Delete(key []byte) error {
	panic("not supported")
}
```

This pattern is used in geth

Reply:
` Get(key []byte) ([]byte, error)` and `Has(key []byte) (bool, error)` also have to be implemented then, to satisfy the `KeyValueReader` required by `VerifyProof`. And at that point I'm writing the same thing, but without the map of memorydb, making lookups less light.

Reply:
Updated the proof verification based on comments from you/josh. The proof is now verified in isolation with `trie.VerifyProof`, but using the db from geth to avoid re-implementing the Get/Has functionality.